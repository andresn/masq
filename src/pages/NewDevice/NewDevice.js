import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { Chevron } from 'icons'
import { Button, QRCode } from 'components'

import './NewDevice.css'

export default class NewDevice extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      image: '',
      lastname: '',
      firstname: '',
      username: ''
    }
  }

  onChange (field, event) {
    this.setState({[field]: event.target.value})
  }

  onImageChange (event) {
    const reader = new window.FileReader()
    const file = event.target.files[0]
    reader.addEventListener('load', () => {
      this.setState({image: reader.result})
    })
    reader.readAsDataURL(file)
  }

  render () {
    const { link, image } = this.props
    const style = {color: 'var(--main-highlight-color)'}

    return (
      <div className='NewDevice'>
        <Link to='devices' className='goback'>
          <Chevron transform='rotate(90)' color='#458bf8' />
          <h3>Go back to the devices list</h3>
        </Link>
        <div className='container'>
          <QRCode image={image} />
          <p className='link'>{link}</p>
          <p>Scan your QR Code on the devices you want to sync with</p>
          <p style={style}>OR</p>
          <p>Receive a link by email</p>
          <Button secondary label='Send link by email' />
        </div>
      </div>
    )
  }
}

NewDevice.propTypes = {
  link: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
}
