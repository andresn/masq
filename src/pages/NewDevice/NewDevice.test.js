import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import NewDevice from './NewDevice'

it('renders Register page', () => {
  const component = renderer.create(
    <MemoryRouter>
      <NewDevice link='QWA.NT/0BJ8ZX' image='http://www.datafakegenerator.com/temp/test409236e0431d1d4dcf66ffba56d9d43c.png' />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
