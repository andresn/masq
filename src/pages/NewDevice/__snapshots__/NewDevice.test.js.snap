// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Register page 1`] = `
<div
  className="NewDevice"
>
  <a
    className="goback"
    href="devices"
    onClick={[Function]}
  >
    <svg
      color="#458bf8"
      height={10}
      transform="rotate(90)"
      viewBox="0 0 14 10"
      width={14}
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <defs>
        <path
          d="M13.7 1.7l-6 6c-.2.2-.4.3-.7.3-.3 0-.5-.1-.7-.3l-6-6C-.1 1.3-.1.7.3.3c.4-.4 1-.4 1.4 0L7 5.6 12.3.3c.4-.4 1-.4 1.4 0 .4.4.4 1 0 1.4z"
          id="chevron-svg"
        />
      </defs>
      <use
        fill="currentColor"
        fillRule="nonzero"
        xlinkHref="#chevron-svg"
      />
    </svg>
    <h3>
      Go back to the devices list
    </h3>
  </a>
  <div
    className="container"
  >
    <div
      className="QRCode"
      style={
        Object {
          "backgroundImage": "url(undefined)",
        }
      }
    />
    <p
      className="link"
    />
    <p>
      Scan your QR Code on the devices you want to sync with
    </p>
    <p
      style={
        Object {
          "color": "var(--main-highlight-color)",
        }
      }
    >
      OR
    </p>
    <p>
      Receive a link by email
    </p>
    <button
      className="Button secondary"
      onClick={undefined}
    >
      Send link by email
    </button>
  </div>
</div>
`;
